### ISSUE TITLE
HELP GUYS ITS REALLY IMPORTANT!!

### ISSUE THREAD

**Issue by SchmerzensReich** [pytorch/pytorch#116080]
### 🐛 Describe the bug

just wanna train my laura and i get the same error everytime, even when i go down with my parameters to relieve my gpu.

not using any commands to start the process but starting it in the gui. 

it starts with  error 10049 something about torch but i cant seem to find another one with the same bug. 

would be very appreciate any soulutions.

### Error logs

[W ..\torch\csrc\distributed\c10d\socket.cpp:601] [c10d] The client socket has failed to connect to [DESKTOP-7I8U3NU]:29500 (system error: 10049 - Die angeforderte Adresse ist in diesem Kontext ung³ltig.).
[W ..\torch\csrc\distributed\c10d\socket.cpp:601] [c10d] The client socket has failed to connect to [DESKTOP-7I8U3NU]:29500 (system error: 10049 - Die angeforderte Adresse ist in diesem Kontext ung³ltig.).
[W ..\torch\csrc\distributed\c10d\socket.cpp:601] [c10d] The client socket has failed to connect to [DESKTOP-7I8U3NU]:29500 (system error: 10049 - Die angeforderte Adresse ist in diesem Kontext ung³ltig.).
[W ..\torch\csrc\distributed\c10d\socket.cpp:601] [c10d] The client socket has failed to connect to [DESKTOP-7I8U3NU]:29500 (system error: 10049 - Die angeforderte Adresse ist in diesem Kontext ung³ltig.).
Traceback (most recent call last):
  File "C:\Users\noName\source\repos\kohya_ss\sdxl_train_network.py", line 185, in <module>
Traceback (most recent call last):
      File "C:\Users\noName\source\repos\kohya_ss\sdxl_train_network.py", line 185, in <module>
trainer.train(args)
trainer.train(args)  File "C:\Users\noName\source\repos\kohya_ss\train_network.py", line 220, in train

      File "C:\Users\noName\source\repos\kohya_ss\train_network.py", line 220, in train
accelerator = train_util.prepare_accelerator(args)
accelerator = train_util.prepare_accelerator(args)  File "C:\Users\noName\source\repos\kohya_ss\library\train_util.py", line 3826, in prepare_accelerator

  File "C:\Users\noName\source\repos\kohya_ss\library\train_util.py", line 3826, in prepare_accelerator
    accelerator = Accelerator(
  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\accelerate\accelerator.py", line 369, in __init__
        accelerator = Accelerator(self.state = AcceleratorState(

  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\accelerate\accelerator.py", line 369, in __init__
  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\accelerate\state.py", line 732, in __init__
    self.state = AcceleratorState(
  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\accelerate\state.py", line 732, in __init__
PartialState(cpu, **kwargs)
PartialState(cpu, **kwargs)  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\accelerate\state.py", line 202, in __init__

      File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\accelerate\state.py", line 202, in __init__
torch.distributed.init_process_group(backend=self.backend, **kwargs)
torch.distributed.init_process_group(backend=self.backend, **kwargs)  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\torch\distributed\distributed_c10d.py", line 907, in init_process_group

  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\torch\distributed\distributed_c10d.py", line 907, in init_process_group
    default_pg = _new_process_group_helper(
      File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\torch\distributed\distributed_c10d.py", line 1013, in _new_process_group_helper
default_pg = _new_process_group_helper(
      File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\torch\distributed\distributed_c10d.py", line 1013, in _new_process_group_helper
raise RuntimeError("Distributed package doesn't have NCCL " "built in")
raise RuntimeError("Distributed package doesn't have NCCL " "built in")RuntimeError
: RuntimeErrorDistributed package doesn't have NCCL built in:
Distributed package doesn't have NCCL built in
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 0 (pid: 14708) of binary: C:\Users\noName\Source\Repos\kohya_ss\venv\Scripts\python.exe
Traceback (most recent call last):
  File "C:\Users\noName\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\noName\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\noName\Source\Repos\kohya_ss\venv\Scripts\accelerate.exe\__main__.py", line 7, in <module>
  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\accelerate\commands\accelerate_cli.py", line 47, in main
    args.func(args)
  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py", line 977, in launch_command
    multi_gpu_launcher(args)
  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\accelerate\commands\launch.py", line 646, in multi_gpu_launcher
    distrib_run.run(args)
  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\torch\distributed\run.py", line 785, in run
    elastic_launch(
  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\torch\distributed\launcher\api.py", line 134, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "C:\Users\noName\Source\Repos\kohya_ss\venv\lib\site-packages\torch\distributed\launcher\api.py", line 250, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError:
============================================================
./sdxl_train_network.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2023-12-19_04:16:50
  host      : DESKTOP-7I8U3NU
  rank      : 1 (local_rank: 1)
  exitcode  : 1 (pid: 16628)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2023-12-19_04:16:50
  host      : DESKTOP-7I8U3NU
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 14708)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================

### Minified repro

============================================================
./sdxl_train_network.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2023-12-19_04:06:24
  host      : DESKTOP-7I8U3NU
  rank      : 1 (local_rank: 1)
  exitcode  : 1 (pid: 6660)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2023-12-19_04:06:24
  host      : DESKTOP-7I8U3NU
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 16028)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html

### Versions

Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Home
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.12.1 (tags/v3.12.1:2305ca5, Dec  7 2023, 22:03:25) [MSC v.1937 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19045-SP0
Is CUDA available: N/A
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration:
GPU 0: NVIDIA GeForce GTX 1080
GPU 1: NVIDIA GeForce RTX 3060

Nvidia driver version: 546.33
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: N/A

CPU:
Architecture=9
CurrentClockSpeed=3600
DeviceID=CPU0
Family=205
L2CacheSize=1536
L2CacheSpeed=
Manufacturer=GenuineIntel
MaxClockSpeed=3600
Name=Intel(R) Core(TM) i5-8600K CPU @ 3.60GHz
ProcessorType=3
Revision=

Versions of relevant libraries:
[pip3] No relevant packages
[conda] Could not collect

cc @ezyang @msaroufim @wconstab @bdhirsh @anijain2305 @zou3519

**Comment by MichaelHillier** [comment]
First, I suggest writing a better post. Title in all caps is obnoxious.

This could be your problem (related to distributed training): 
"As of PyTorch v1.8, Windows supports all collective communications backend but NCCL" [https://pytorch.org/docs/stable/distributed.html](url)
Try gloo backend.

**Comment by voznesenskym** [comment]
Title is clickbait, content of issue are hard to parse, formatted poorly.
